### ISSUE TITLE
Use otoole with custom OSeMOSYS parameters

### ISSUE THREAD

**Issue by willu47** [OSeMOSYS/otoole#56]
If you modify the name of a parameter or set, or add a new parameter or set in an OSeMOSYS model which you wish to use with otoole, there is currently no way for the user to edit the `config.yaml` file.

Suggest:

- [x] adding `--config` flag to the command line
- [x] `setup` command to write the configuration to a file in a user directory
- [x] add documentation on how to add or remove parameters #113 #107
- [x] validate user config #133

**Comment by HauHe** [comment]
I have created a modified version of the OSeMOSYS code in which I added parameter.
To be able to still use otoole with the data file for the modified OSeMOSYS code I did took the following steps:
- cloned the repository
- created a new conda environment
- from the root of the repository I used the command `python setup.py develop` to install a development version of otoole in the environment created in the previous step. Using otoole in this environment directly considers all changes done in the repository
In my specific case I added several parameter to the OSeMOSYS code. To enable otoole to deal with datafiles containing the additional parameter the following two files had to be updated:
- src/otoole/preprocess/config.yaml
- src/otoole/preprocess/datapackage.json
In these files all parameter are listed. And new parameter need to be added to avoid errors.
